src/registrar.js

```diff
@@ -155,13 +155,14 @@ const getPermanentEntry = async label => {
     }
     // This is used for old registrar to figure out when the name can be migrated.
     obj.migrationLockPeriod = parseInt(await Registrar.MIGRATION_LOCK_PERIOD())
```

Comment:
Why do we call these for each entry? Couldn't we fetch them once somewhere, and cache them, to save on unnecessary round-trips?